# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../etoile/compile.patch
# Copyright (C) 2007 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- Etoile-0.2/Frameworks/PopplerKit/bindings/poppler.cc.vanilla	2007-11-19 20:13:10.000000000 +0100
+++ Etoile-0.2/Frameworks/PopplerKit/bindings/poppler.cc	2007-11-19 20:13:26.000000000 +0100
@@ -123,7 +123,7 @@
          fflush(stderr);
       }
 
-      globalParams = new GlobalParams(NULL);
+      globalParams = new GlobalParams();
 #ifdef POPPLER_0_4
       globalParams->setupBaseFontsFc(NULL);
 #endif
--- Etoile-0.2/Frameworks/XWindowServerKit/GNUmakefile.vanilla	2007-11-19 20:39:53.000000000 +0100
+++ Etoile-0.2/Frameworks/XWindowServerKit/GNUmakefile	2007-11-19 20:40:01.000000000 +0100
@@ -19,7 +19,7 @@
 XWindowServerKit_RESOURCE_FILES = ScriptingInfo.plist
 
 ADDITIONAL_INCLUDE_DIRS += -I/usr/X11R6/include
-ADDITIONAL_LIB_DIRS += -L/usr/X11R6/lib -lX11
+ADDITIONAL_LIB_DIRS += $(shell  pkg-config x11 --libs)
 
 include $(GNUSTEP_MAKEFILES)/aggregate.make
 include ../../etoile.make
--- Etoile-0.2/Bundles/EtoileWildMenus/NSMenuView+Hackery.m.vanilla	2007-11-19 20:18:23.000000000 +0100
+++ Etoile-0.2/Bundles/EtoileWildMenus/NSMenuView+Hackery.m	2007-11-19 20:18:45.000000000 +0100
@@ -25,6 +25,7 @@
 
 #import <AppKit/NSApplication.h>
 #import <AppKit/PSOperators.h>
+#import <AppKit/NSEvent.h>
 
 #import "EtoileMenuTitleView.h"
 #import "MenuBarHeight.h"
--- Etoile-0.2/Bundles/GNUmakefile.vanilla	2007-11-19 20:23:46.000000000 +0100
+++ Etoile-0.2/Bundles/GNUmakefile	2007-11-19 20:43:00.000000000 +0100
@@ -57,7 +57,7 @@
 endif
 
 ifeq ($(volumecontrolmenulet), yes)
-    SUBPROJECTS += VolumeControlMenulet
+    #SUBPROJECTS += VolumeControlMenulet
 endif
 
 include $(GNUSTEP_MAKEFILES)/aggregate.make
--- Etoile-0.2/Services/Private/Idle/GNUmakefile.vanilla	2007-11-19 20:45:26.000000000 +0100
+++ Etoile-0.2/Services/Private/Idle/GNUmakefile	2007-11-19 20:46:46.000000000 +0100
@@ -25,7 +25,7 @@
 
 ADDITIONAL_OBJCFLAGS = -std=c99 -g -Werror
 ADDITIONAL_INCLUDE_DIRS += -I/usr/X11R6/include
-ADDITIONAL_LIB_DIRS += -L/usr/X11R6/lib -lX11 -lXss
+ADDITIONAL_LIB_DIRS += $(shell pkg-config x11 --libs) -lXss
 
 include $(GNUSTEP_MAKEFILES)/tool.make
 -include ../../../etoile.make
--- Etoile-0.2/Services/Private/Corner/GNUmakefile.vanilla	2007-11-19 20:45:07.000000000 +0100
+++ Etoile-0.2/Services/Private/Corner/GNUmakefile	2007-11-19 20:45:14.000000000 +0100
@@ -24,7 +24,7 @@
 
 ADDITIONAL_OBJCFLAGS = -std=c99 -g -Werror
 ADDITIONAL_INCLUDE_DIRS += -I/usr/X11R6/include
-ADDITIONAL_LIB_DIRS += -L/usr/X11R6/lib -lX11
+ADDITIONAL_LIB_DIRS += $(shell  pkg-config x11 --libs)
 ADDITIONAL_LDFLAGS += -lStepTalk -g
 
 include $(GNUSTEP_MAKEFILES)/tool.make
--- Etoile-0.2/Services/Private/AZBackground/GNUmakefile.vanilla	2007-11-19 20:43:28.000000000 +0100
+++ Etoile-0.2/Services/Private/AZBackground/GNUmakefile	2007-11-19 20:43:41.000000000 +0100
@@ -14,7 +14,7 @@
 ADDITIONAL_INCLUDE_DIRS += -I/usr/include -I/usr/X11R6/include 
 
 ADDITIONAL_LIB_DIRS += \
-	-L/usr/lib -L/usr/X11R6/lib -lX11 -lXWindowServerKit 
+	-L/usr/lib $(shell pkg-config x11 --libs) -lXWindowServerKit 
 
 ADDITIONAL_GUI_LIBS += 
 
--- Etoile-0.2/Services/Private/Login/GNUmakefile.vanilla	2007-11-19 20:44:47.000000000 +0100
+++ Etoile-0.2/Services/Private/Login/GNUmakefile	2007-11-19 20:44:54.000000000 +0100
@@ -22,7 +22,7 @@
 
 Login_MAIN_MODEL_FILE = login.gorm
 
-ADDITIONAL_LDFLAGS = -lX11
+ADDITIONAL_LDFLAGS = $(shell  pkg-config x11 --libs)
 
 include $(GNUSTEP_MAKEFILES)/application.make
 include ../../../etoile.make
--- Etoile-0.2/Services/Private/AZSwitch/GNUmakefile.vanilla	2007-11-19 20:47:02.000000000 +0100
+++ Etoile-0.2/Services/Private/AZSwitch/GNUmakefile	2007-11-19 20:47:28.000000000 +0100
@@ -16,7 +16,7 @@
 	-I/usr/X11R6/include 
 
 ADDITIONAL_LIB_DIRS += \
-	-L/usr/X11R6/lib -lX11 \
+	$(shell pkg-config x11 --libs) \
 	-lXWindowServerKit \
 	-lEtoileUI
 
--- Etoile-0.2/Services/Private/AZDock/GNUmakefile.vanilla	2007-11-19 20:44:13.000000000 +0100
+++ Etoile-0.2/Services/Private/AZDock/GNUmakefile	2007-11-19 20:44:20.000000000 +0100
@@ -25,7 +25,7 @@
 ADDITIONAL_INCLUDE_DIRS += -I/usr/include -I/usr/X11R6/include
 
 ADDITIONAL_LIB_DIRS += \
-	-L/usr/lib -L/usr/X11R6/lib
+	-L/usr/lib $(shell  pkg-config x11 --libs)
 
 ADDITIONAL_GUI_LIBS += \
 	-lX11 \
